### PR TITLE
Upload via clipboard

### PR DESCRIPTION
**Pull request by broskoski**

### CODE REVIEW

File: src/v2/components/AddBlock/components/AddBlockPasteUploader/index.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  channelId,
+}) => {
+  const [mode, setMode] = useState('resting')
+  const [fileUrl, setFileUrl] = useState(null)
+
+  usePasteListener({
+    onPaste: url => {
+      setMode('active')
+      setFileUrl(url)
+    },
```
Comment (dzucconi) [inline src/v2/components/AddBlock/components/AddBlockPasteUploader/index.tsx]: I'd just avoid inlining callbacks in general and make sure you use useCallback

File: src/v2/components/AddBlock/components/AddBlockPasteUploader/index.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  })
+
+  const finishUpload = () => {
+    setFileUrl(null)
```
Comment (dzucconi) [inline src/v2/components/AddBlock/components/AddBlockPasteUploader/index.tsx]: ditto

**Comment by broskoski** [comment]
clipboard-paster

File: src/v2/components/AddBlock/components/AddBlockPasteUploader/index.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  const onUpload = useCallback(
+    ({ url: value }) => {
+      setFileUrl(null)
+      setMode('resting')
+      return createBlock({ variables: { channel_id: channelId, value } })
+        .then(({ data: { create_block: { block } } }) => onAddBlock(block))
+        .catch(err => {
+          console.error(err)
```
Comment (dzucconi) [inline src/v2/components/AddBlock/components/AddBlockPasteUploader/index.tsx]: Can we bubble up any kind of user feedback? It's unfortunate we don't have a toast-like error UI we could use.

File: src/v2/hooks/usePasteListener/index.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      transformImages(e.clipboardData, acceptedFiles, (url: string) => {
+        setFileUrl(url)
+        onPaste(url)
+      })
+    }
+
+    el.addEventListener('paste', pasteHandler)
+
+    return () => {
+      el.removeEventListener('paste', pasteHandler)
```
Comment (dzucconi) [inline src/v2/hooks/usePasteListener/index.tsx]: ✔️

File: src/v2/components/AddBlock/components/AddBlockPasteUploader/index.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  right: 0;
+  bottom: 0;
+  left: 0;
+  z-index: 3;
+`
+
+interface AddBlockPasteUploader {
+  onAddBlock: (props: any) => any
+  channelId: string | number
+  createBlock: (props: any) => Promise<any>
```
Comment (dzucconi) [inline src/v2/components/AddBlock/components/AddBlockPasteUploader/index.tsx]: Would avoid all these anys. I know there's a lot of them but just get rid of them. Hover to find out return types at the call sites and just paste em around.

File: src/v2/components/AddBlock/components/AddBlockPasteUploader/index.tsx (hunk 3 above)
Comment (broskoski) [reply src/v2/components/AddBlock/components/AddBlockPasteUploader/index.tsx]: Yeah, true. Let me see what you do in the drop zone uploader.

File: src/v2/components/AddBlock/components/AddBlockPasteUploader/index.tsx (hunk 5 above)
Comment (broskoski) [reply src/v2/components/AddBlock/components/AddBlockPasteUploader/index.tsx]: Now I remember why I did that:
Screen Shot 2019-10-07 at 1 34 40 PM

Will fix.

Comment (dzucconi) [reply src/v2/components/AddBlock/components/AddBlockPasteUploader/index.tsx]: Yeah, I suspect most of my TypeScript from this era isn't exactly exemplar.

**Pull request merged**